app/models/loan.rb

```diff
@@ -3,4 +3,19 @@ class Loan < ActiveRecord::Base
   belongs_to :order
   validates :order_id, presence: true
   validates :project_id, presence: true
+  validates :amount, presence: true, numericality:
+    {
+      greater_than_or_equal_to: 1000,
+      less_than_or_equal_to: :project_amount_needed
+    }
+
+  private
+
+  def project_amount_needed
+    if project.nil?
+      1000
```

Comment:
I welcome suggestions on how to fix this, but with the tests in test/models/loan_test.rb, I tried many different combinations, and this was the only way I figured out to test attributes with the validations in place.